gremlin-core/src/main/java/org/apache/tinkerpop/gremlin/process/traversal/step/map/GroupStep.java

```diff
@@ -79,6 +79,14 @@ public void modulateBy(final Traversal.Admin<?, ?> kvTraversal) {
     }
 
     @Override
+    public void replaceLocalChild(final Traversal.Admin<?, ?> oldTraversal, final Traversal.Admin<?, ?> newTraversal) {
+        if (null != this.keyTraversal && this.keyTraversal.equals(oldTraversal))
+            this.keyTraversal = this.integrateChild(newTraversal);
+        else if (null != this.valueTraversal && this.valueTraversal.equals(oldTraversal))
```

Comment:
There is going to be more complications here due to barrier handling. `GroupStep` does some pre-processing on the `valueTraversal`.

https://github.com/apache/tinkerpop/blob/c5a1bdebf02538dda19fa73c26e1aa350aa8c32f/gremlin-core/src/main/java/org/apache/tinkerpop/gremlin/process/traversal/step/map/GroupStep.java#L71-L75